primers/pcr/pcr.go

```diff
@@ -0,0 +1,138 @@
+package pcr
+
+import (
+	"github.com/TimothyStiles/poly/primers"
+	"github.com/TimothyStiles/poly/transform"
+	"index/suffixarray"
+	"sort"
+	"strings"
+)
+
+// https://doi.org/10.1089/dna.1994.13.75
+var minimalPrimerLength int = 15
+
+func DesignPrimersWithOverhangs(sequence, forwardOverhang, reverseOverhang string, targetTm float64) (string, string) {
+	sequence = strings.ToUpper(sequence)
+	forwardPrimer := sequence[0:minimalPrimerLength]
+	for i := 0; primers.MeltingTemp(forwardPrimer) < targetTm; i++ {
+		forwardPrimer = sequence[0 : minimalPrimerLength+i]
+	}
+	reversePrimer := transform.ReverseComplement(sequence[len(sequence)-minimalPrimerLength:])
+	for i := 0; primers.MeltingTemp(reversePrimer) < targetTm; i++ {
+		reversePrimer = transform.ReverseComplement(sequence[len(sequence)-(minimalPrimerLength+i):])
+	}
+
+	// Add overhangs to primer
+	forwardPrimer = forwardOverhang + forwardPrimer
+	reversePrimer = transform.ReverseComplement(reverseOverhang) + reversePrimer
+
+	return forwardPrimer, reversePrimer
+}
+
+func DesignPrimers(sequence string, targetTm float64) (string, string) {
+	return DesignPrimersWithOverhangs(sequence, "", "", targetTm)
+}
+
```

Comment:
Needs comment for docs.

Reply:
This also needs more in-line context. Regular devs won't know anything about forward and reverse locations, minimal length, melting temp, overhangs or even what PCR is.